### PR TITLE
Feature/bug 75299

### PR DESCRIPTION
**Pull request by dotcom9**
Fix stylesheet to spec

### CODE REVIEW

**Comment by vrojastpr** [comment]
try adding the following to the `buttons.mdx` file


```
### Disabled

<Playground>
	<div style={{ background: '#e1e1e1' }}>
		<Button cfg={{ mr: 3 }} disabled>
			Click me
		</Button>
		<Button cfg={{ mr: 3 }} appearance="secondary" disabled>
			Click me
		</Button>
		<Button cfg={{ mr: 3, my: 3 }} intent="warning" disabled>
			Click me
		</Button>
	</div>
</Playground>
```

and you will see that the styles for the `disabled` state are not how they should according to https://dev.azure.com/thepensionsregulator/TPR/_workitems/edit/75299

File: packages/core/src/components/buttons/buttons.module.scss [hunk 1]
```diff
@@ -10,53 +10,50 @@
...
 	border-radius: 8px;
 
-	&:hover {
-		background: $colors-neutral-a1 !important;
-	}
-
 	&:focus {
 		background: $colors-warning-1!important;
-		color: $colors-neutral-8;
+		color: white;
```
Comment (HarryAmmon) [inline packages/core/src/components/buttons/buttons.module.scss:45]: Could instances of `white` be changed to the white stored in `variables.scss` to ensure we are using the same white throughout. Its unlikely to change but if it does in the future the color will stay consistent throughout.

**Review by HarryAmmon — CHANGES_REQUESTED** [review]

File: packages/core/src/components/buttons/buttons.module.scss (hunk 1 above)
Comment (dotcom9) [reply packages/core/src/components/buttons/buttons.module.scss:45]: Well spotted. White was the wrong colour here anyway!

**Review by HarryAmmon — APPROVED** [review]

File: packages/core/src/components/buttons/buttons.module.scss [hunk 2]
```diff
@@ -9,54 +9,47 @@
...
-
-.appearance-primary:disabled {
-	cursor: not-allowed !important;
-	background: $colors-primary-a1;
 
-	&:hover {
-		background: $colors-neutral-6 !important;
+	&:disabled {
+		cursor: not-allowed !important;
+		background: $colors-neutral-6;
```
Comment (vrojastpr) [inline packages/core/src/components/buttons/buttons.module.scss:29]: shouldn't this color be `colors-primary-a1` ? same for disabled->hover?

File: packages/core/src/components/buttons/buttons.module.scss [hunk 3]
```diff
@@ -104,22 +126,22 @@
...
-	background: $colors-danger-1;
 
-	&:hover {
-		background: $colors-neutral-6 !important;
+	&:disabled {
+		cursor: not-allowed !important;
+		background: $colors-danger-1;
+	
+		&:hover {
+			background: $colors-neutral-6 !important;
```
Comment (vrojastpr) [inline packages/core/src/components/buttons/buttons.module.scss:142]: probably this one should stay as `$colors-danger-1` when `disabled->hover`

File: packages/core/src/components/buttons/buttons.module.scss [hunk 4]
```diff
@@ -9,54 +9,47 @@
...
 
-.appearance-secondary:disabled {
-	cursor: not-allowed !important;
-	background: $colors-primary-1;
-
-	&:hover {
-		background: $colors-neutral-6 !important;
+	&:disabled {
+		cursor: not-allowed !important;
+		background: $colors-primary-1!important;
```
Comment (vrojastpr) [inline packages/core/src/components/buttons/buttons.module.scss:52]: probably need to add the `disabled->hover` state to keep the `background: $colors-primary-1`, currently is changing

File: packages/core/src/components/buttons/buttons.module.scss (hunk 2 above)
Comment (dotcom9) [reply packages/core/src/components/buttons/buttons.module.scss:29]: yep

File: packages/core/src/components/buttons/buttons.module.scss (hunk 4 above)
Comment (dotcom9) [reply packages/core/src/components/buttons/buttons.module.scss:52]: Fixed in PR #281

File: packages/core/src/components/buttons/buttons.module.scss (hunk 3 above)
Comment (dotcom9) [reply packages/core/src/components/buttons/buttons.module.scss:142]: Fixed in PR #281

**Pull request merged**